src/pages/contract/_components/ReadContractCodeRow.tsx

```diff
@@ -0,0 +1,172 @@
+import GradientCardContainer from "@components/commons/GradientCardContainer";
+import { Dispatch, SetStateAction, useEffect, useRef, useState } from "react";
+import { FiCopy, FiShare2, FiMaximize, FiMinimize } from "react-icons/fi";
+import { sleep } from "shared/sleep";
+import { CodeBlock } from "react-code-blocks";
+import clsx from "clsx";
+import Tooltip from "@components/commons/Tooltip";
+
+export default function ReadContractCodeRow({
+  fileName,
+  code,
+  length,
+  index,
+}: {
+  fileName: string;
+  code: string;
+  length: number;
+  index: number;
+}) {
+  const [isCodeCopied, setIsCodeCopied] = useState(false);
+  const [isCodeExpanded, setIsCodeExpanded] = useState(false);
+  const [isPermaLinkClicked, setIsPermaLinkClicked] = useState(false);
+  const count = useRef(0);
+  const [isHovering, setIsHovering] = useState(false);
+
+  const handleMouseEnter = () => {
+    setIsHovering(true);
+  };
+
+  const handleMouseLeave = () => {
+    setIsHovering(false);
+  };
+
+  useEffect(() => {
+    count.current += 1;
```

Comment:
is the `count` only to be used to check when the component first load?
If yes, consider replacing with a boolean flag to avoid `count` from getting infinitely large

Reply:
updated to using a boolean